sources/Context.cpp

```diff
@@ -202,11 +271,53 @@ Context::execute() {
   // Since DFL traces are persistent, they can be re-used after simulation is performed outside this function
   LOG(info) << MESS(SimulateInfo, basename_) << LOG_ENDL;
 
-  auto simu = boost::make_shared<DYN::Simulation>(jobEntry_, simu_context, networkManager_.dataInterface());
-  simu->init();
-  simu->simulate();
-  simu->terminate();
-  simu->clean();
+  switch (def_.simulationKind) {
+  case SimulationKind::STEADY_STATE_CALCULATION: {
+    // For a power flow calculation it is ok to directly run here a single simulation
+    auto simu = boost::make_shared<DYN::Simulation>(jobEntry_, simu_context, networkManager_.dataInterface());
+    simu->init();
+    simu->simulate();
+    simu->terminate();
+    simu->clean();
+    break;
+  }
+  case SimulationKind::SECURITY_ANALYSIS:
+    executeSecurityAnalysis();
+    break;
+  }
+}
+
+void
+Context::executeSecurityAnalysis() {
+  // For security analysis we run multiple simulations using dynawo-algorithms
+  // Create one scenario for the base case and one scenario for each contingency
+  auto scenarios = boost::make_shared<DYNAlgorithms::Scenarios>();
+  scenarios->setJobsFile(jobEntry_->getName() + ".jobs");
+  auto baseCase = boost::make_shared<DYNAlgorithms::Scenario>();
+  baseCase->setId("BaseCase");
+  scenarios->addScenario(baseCase);
+  for (const auto& c : contingencies_.definitions()) {
+    if (contingencies_.isValidForSimulation(*c)) {
+      auto scenario = boost::make_shared<DYNAlgorithms::Scenario>();
+      scenario->setId(c->id);
+      scenario->setDydFile(basename_ + "-" + c->id + ".dyd");
+      scenarios->addScenario(scenario);
+      LOG(info) << MESS(ContingencySimulationDefined, c->id) << LOG_ENDL;
+    }
+  }
+  // Use dynawo-algorithms Systematic Analysis Launcher to simulate all the scenarios
+  auto multipleJobs = multipleJobs::MultipleJobsFactory::newInstance();
+  multipleJobs->setScenarios(scenarios);
+  auto saLauncher = boost::make_shared<DYNAlgorithms::SystematicAnalysisLauncher>();
+  saLauncher->setMultipleJobs(multipleJobs);
+  saLauncher->setOutputFile("sa.zip");
+  saLauncher->setDirectory(config_.outputDir().native());
```

Comment:
Use generic_string() instead of native : safer for the type and compliant with rest of the code

Reply:
Done